lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -172,19 +187,38 @@ data StakePoolLayer = StakePoolLayer
         -- Exclude all pools that retired in or before this epoch.
         -> Coin
         -> ExceptT ErrListPools IO [Api.ApiStakePool]
+
+    , putSettings :: Settings -> IO ()
+
+    , getSettings :: IO Settings
+
+    , clearMetadataTable :: IO ()
+
+    , metadataThread :: TVar ThreadId
```

Comment:
I don't think we want to be exposing such implementation details in the stake pool layer interface.